site/blog/2022-10-28-optimize-s3-endpoints.mdx

```diff
@@ -0,0 +1,121 @@
+
+<p>
+  <a href="https://app.iasql.com/#/button/Check S3 endpoints/--%20Installing%20the%20needed%20modules%0ASELECT%20iasql_install%28%27aws_s3%27%29%3B%0ASELECT%20iasql_install%28%27aws_vpc%27%29%3B%0A--%20Sync%20the%20existing%20data%0Aselect%20%2A%20from%20iasql_sync%28%29%3B%0A--%20Perform%20the%20query%20for%20endpoints%0Aselect%20bucket.region%2C%20vpc.is_default%2C%20vpc.cidr_block%2C%0A%28select%20count%28%2A%29%3E0%20from%20endpoint_gateway%20where%20endpoint_gateway.region%3Dbucket.region%20and%20service%3D%27s3%27%20and%20endpoint_gateway.vpc_id%3Dvpc.id%29%20as%20has_endpoint_gateway%2C%0A%28select%20count%28%2A%29%3E0%20from%20endpoint_interface%20where%20endpoint_interface.region%3Dbucket.region%20and%20service%3D%27s3%27%20and%20endpoint_interface.vpc_id%3Dvpc.id%29%20as%20has_endpoint_interface%0Afrom%20bucket%20left%20outer%20join%20vpc%20on%20vpc.region%3Dbucket.region%3B">
+    <button class="button button--lg button--primary">Check your endpoints</button>
+  </a>
+</p>
+
+Have you found missing endpoints? No problem, IaSQL can generate them for you. You can create two different
+types of endpoints: gateway and interface. We're going to describe their main features and you could create
+the desired type in an automated way.
+
+## Endpoint interfaces
+
+And endpoint interface is an elastic network interface with a private IP address that serves as an entry
+point for traffic destined to a supported AWS service. As they use ENIs, they can benefit of security groups
+to control traffic. Policies can also be added to control access to those endpoints.
+
+They are regional, meaning they can only be accessed by the same region they are created. Multi-region is
+possible using VPC peering, so resources in one region can be accessed from others. They only support IPv4
+TCP traffic.
+
+An interface endpoint (except S3 interface endpoint) has corresponding private DNS hostnames, that can be
+used to access the resources.
+
+Endpoint interfaces cover a wide range of services such as S3, Lambda, API Gateway, etc...
+
+<p>
+  <a href="https://app.iasql.com/#/button/Add S3 endpoint interfaces/--%20Inserts%20the%20missing%20endpoints%0Ainsert%20into%20endpoint_interface%28region%2C%20vpc_id%2C%20service%29%20select%20bucket.region%2C%20vpc.id%2C%20%27s3%27%20from%20bucket%20inner%20join%20vpc%0Aon%20bucket.region%3Dvpc.region%20where%20not%20exists%20%28select%20id%20from%20endpoint_interface%20where%20endpoint_interface.region%3Dbucket.region%20and%20endpoint_interface.vpc_id%3Dvpc.id%29%3B%0A--%20Preview%20the%20changes%0Aselect%20%2A%20from%20iasql_preview_apply%28%29%3B%0A--%20Apply%20the%20changes%0A--select%20%2A%20from%20iasql_apply%28%29%3B">
+    <button class="button button--lg button--primary">Add missing endpoint interfaces</button>
+  </a>
+</p>
+
+By clicking on the button, our engine will auto-generate the missing endpoint interfaces and will allow you
+to preview the changes to be applied on your cloud.
+
```

Comment:
Maybe we can add here at the end a link to the AWS docs in endpoint interfaces? Like `Get more detail about ...`